File: services/ats-system-checker/src/utils/logging_config.py
"""
Logging Configuration Utilities for ATS System
"""

import logging
import os
from typing import Dict


def setup_logging(log_level: str = "INFO", log_file: str = "ats_system.log") -> None:
    """
    Setup logging configuration for the application
    
    Args:
        log_level: Logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
        log_file: Log file path
    """
    # Log level mapping
    LOG_LEVELS: Dict[str, int] = {
        'DEBUG': logging.DEBUG,
        'INFO': logging.INFO, 
        'WARNING': logging.WARNING,
        'ERROR': logging.ERROR,
        'CRITICAL': logging.CRITICAL
    }
    
    # Get log level, default to INFO if invalid
    level = LOG_LEVELS.get(log_level.upper(), logging.INFO)
    
    # Create logs directory if it doesn't exist
    log_dir = os.path.dirname(log_file) if os.path.dirname(log_file) else "."
    if log_dir != "." and not os.path.exists(log_dir):
        os.makedirs(log_dir, exist_ok=True)
    
    # Configure logging
    logging.basicConfig(
        level=level,
        format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
        handlers=[
            logging.FileHandler(log_file, encoding='utf-8'),
            logging.StreamHandler()
        ]
    )
    
    # Set specific logger levels
    logging.getLogger("uvicorn").setLevel(logging.WARNING)
    logging.getLogger("uvicorn.access").setLevel(logging.WARNING)
    
    logger = logging.getLogger(__name__)
    logger.info(f" Logging configured - Level: {log_level}, File: {log_file}")


def get_logger(name: str) -> logging.Logger:
    """
    Get a logger instance for a specific module
    
    Args:
        name: Logger name (usually __name__)
        
    Returns:
        logging.Logger: Configured logger instance
    """
    return logging.getLogger(name) 

File: services/ats-system-checker/src/utils/logging_config.ts
import fs from "fs";
import path from "path";

const LOG_LEVELS: Record<string, number> = {
  'DEBUG': 0,
  'INFO': 1,
  'WARNING': 2,
  'ERROR': 3,
  'CRITICAL': 4
};

export function setup_logging(log_level: string = "INFO", log_file: string = "ats_system.log"): void {
  const level = LOG_LEVELS[log_level.toUpperCase()] ?? 1;
  
  // Create logs directory if it doesn't exist
  const log_dir = path.dirname(log_file);
  if (log_dir !== "." && !fs.existsSync(log_dir)) {
    fs.mkdirSync(log_dir, { recursive: true });
  }
  
  // Note: In Node.js, we typically use winston or another logging library
  // For now, we'll use console with basic formatting
  console.log(`Logging configured - Level: ${log_level}, File: ${log_file}`);
}

export function get_logger(name: string): Console {
  // Return console for now - can be replaced with winston or another logger
  return console;
}

